Added test for ChangeLogModalController, AppFooterController and PeerSelectController

bartist opened the pull request:
Part of Issue (issuenumber).
This pull request will improve the test scores of the controller.js as follows:


|             | % Statements | % Branch | % Functions | % Lines |
|:-----------:|:------------:|:--------:|-------------|---------|
|  Currently  |      3.5     |     0    |    0.36     |   3.52  |
| After Merge |      5.61    |    1.16  |    2.88     |   5.64  |

This pull request fully tests the ChangeLogModalController, AppFooterController and PeerSelectController.
If a certain style is demanded for creating, then we will adapt the test to conform with the demanded style.
This is also the reason that this pull request tests some of the smaller controllers, in case changes are needed.

bartist committed:
Added test for ChangeLogModalController
This commit changes the test scores as follows

|     | % Statements | % Branch | % Functions | % Lines |
|:---:|:------------:|:--------:|-------------|---------|
| Old |      3.5     |     0    | 0.36        | 3.52    |
| New |     4.22     |   0.36   | 1.08        | 4.24    |

bartist committed:
Added tests for AppFooterController
This commit changes the test scores as follows

|     | % Statements | % Branch | % Functions | % Lines |
|:---:|:------------:|:--------:|-------------|---------|
| Old |     4.22     |   0.36   |    1.08     |   4.24  |
| New |     4.29     |   0.36   |    1.44     |   4.31  |

bartist committed:
Added tests for PeerSelectController
This commit changes the test scores as follows

|     | % Statements | % Branch | % Functions | % Lines |
|:---:|:------------:|:--------:|-------------|---------|
| Old |     4.29     |   0.36   |    1.44     |   4.31  |
| New |     5.61     |   1.16   |    2.88     |   5.64  |

Stapelpanda reviewed (approved):
Looks good, 1 Second should suffice in the Peerselect controller, maybe nice to use a constant for this such that it can easily be changed?

bartist commented:
Great suggestion! 
I will fix that after dinner

bartist committed:
Change Time-out time to variable
Instead of doing time-outs based on a constant, the time-out is based on a variable. This will make changing the time of time-out easier.

kiliangrashoff reviewed (approved):
I think the timeouts in PeerSelectControllerSpec are a code smell, but I do not currently have a suggestion on how to avoid them, so I think we can submit this PR and see if we can improve this later. We may also get feedback when we submit the PR on how to avoid this.
